packages/core/src/util/configNormalizers/ignoreEndpoints.js

```diff
@@ -0,0 +1,121 @@
+ * - Strings or arrays of strings: These are interpreted as methods to ignore.
+ * - Objects: Should include `methods` and/or `endpoints` properties.
+ *
+ * Normalized internal structure:
+ * - { [serviceName: string]: [{ methods: string[], endpoints: string[] }] }
+ *
+ * @param {import('../../tracing').IgnoreEndpoints} ignoreEndpointConfig
+ * @returns {import('../../tracing').IgnoreEndpoints}
+ */
+
+exports.normalizeConfig = function normalizeConfig(ignoreEndpointConfig) {
+  if (!ignoreEndpointConfig || typeof ignoreEndpointConfig !== 'object') {
+    return {};
+  }
+  try {
+    return Object.fromEntries(
+      Object.entries(ignoreEndpointConfig).map(([service, endpointConfigs]) => {
+        if (!Array.isArray(endpointConfigs)) {
+          return [normalizeString(service), []];
+        }
+
+        // Normalize string-based configurations by treating each string as a method name.
+        const methodNames = /** @type {string[]} */ (endpointConfigs.filter(config => typeof config === 'string')).map(
+          normalizeString
+        );
+
+        // Normalize object-based configurations that may include 'methods' and/or 'endpoints'.
+        const advancedConfigs = endpointConfigs
+          .filter(config => typeof config === 'object' && config !== null)
+          .map(config => {
+            // Normalize configuration keys to support case-insensitivity.
+            const normalizedCfg = Object.fromEntries(
+              Object.entries(config).map(([key, value]) => [normalizeString(key), value])
+            );
+            const validConfig = {};
+            if (normalizedCfg.methods) {
+              validConfig.methods = [].concat(normalizedCfg.methods).map(normalizeString);
+            }
+            if (normalizedCfg.endpoints) {
+              validConfig.endpoints = [].concat(normalizedCfg.endpoints).map(normalizeString);
+            }
+            // extend the config with more fields here
+            return Object.keys(validConfig).length ? validConfig : null;
+          })
+          .filter(Boolean);
+
+        // Combine both string-based and object-based configurations.
+        const normalizedConfigs = methodNames.length ? [{ methods: methodNames }, ...advancedConfigs] : advancedConfigs;
+        return [normalizeString(service), normalizedConfigs];
+      })
+    );
+  } catch (error) {
+    logger?.warn('Error when parsing ignore-endpoints configuration', error?.message);
+    return {};
+  }
+};
+
+/**
+ * Parses the `INSTANA_IGNORE_ENDPOINTS` environment variable.
+ * Only basic filtering supported by this env variable.
+ * Expected format: - "service:endpoint1,endpoint2"
+ * @param {string} ignoreEndpointsEnv
+ */
+exports.parseIgnoreEndpointsFromEnv = function parseIgnoreEndpointsFromEnv(ignoreEndpointsEnv) {
+  try {
+    if (!ignoreEndpointsEnv) {
+      return {};
+    }
+    return Object.fromEntries(
+      ignoreEndpointsEnv
+        .split(';')
+        .map(serviceEntry => {
```

Comment:
qs: Why dont we reuse the normalize fn? 🤔 

INSTANA_IGNORE_ENDPOINTS="redis:get;dynamodb:set"

Pseudo code three lines:

```js
let result = ignoreEndpointsEnv.split(';')
result = res.map((x) => {return { [x.split(':')[0]]: x.split(':')[1] }})

// -> [ { redis: 'get' }, { dynamodb: 'set' } ]

return normalizeConfig(result)

Reply:
> if (!serviceName || !endpointList) {

IMO either we check this in "normalizeConfig" if possible.